paths/pythonParser.py

```diff
@@ -0,0 +1,190 @@
+            fileName[:len(fileName) - 4] + 'DiscardedData.txt', "w") as trash:
+        sadness = random.randint(1, 1000000)
+        text = random.randint(0, len(splashText) - 1)
+        lol = random.randint(0, len(splashText) - 1)
+        e1 = random.randint(0, len(splashText) - 1)
+        e2 = random.randint(0, len(splashText) - 1)
+        ree = random.randint(0, len(copy_right) - 1)
+        year = random.randint(1945, 3099)
+        f_str = f.read()
+        f_str.encode('ascii', 'replace')
+        for a in bad_chars:
+            trash.write(a)
+            f_str = f_str.replace(a, '')
+        g = StringIO(f_str)
+        print(" Python Parser is about to Parse in Python")
+        time.sleep(1.5)
+        print(" Parsing in")
+        time.sleep(0.75)
+        print("  3")
+        time.sleep(0.5)
+        print("  2")
+        time.sleep(0.5)
+        print("  1")
+        time.sleep(0.5)
+        print(" Python Parser Parsed in Python Succsesfully")
+        time.sleep(0.25)
+        print(" Python Parser saved the java arrays to:\n " + path + "\\src\\main\\java\\frc\\robot\\routes\\" +
+              fileName[6:len(fileName) - 4] + ".java\n and the graph to: " + path + fileName[6:len(fileName) - 4] + "\\"
+              + 'GraphData.txt\n and the discarded data to: ' + path + "\\" + fileName[6:len(fileName) - 4] +
+              'DiscardedData.txt')
+        trash.write(" \n")
+        for line in g.readlines():
+            if line.startswith("-") or line.startswith(" ") or line[0].isdigit():
+                for entry in line.strip().split(" "):
+                    try:
+                        float(entry)
+                    except ValueError:
+                        trash.write(entry + " \n")
+                        continue
+                    if i == 0:
+                        velocities.append(entry)
+                    if i == 1:
+                        rotations.append(entry)
+                    if i == 2:
+                        frontLeftEncoderPositions.append(entry)
+                    if i == 3:
+                        frontRightEncoderPositions.append(entry)
+                    if i == 4:
+                        backLeftEncoderPositions.append(entry)
+                    if i == 5:
+                        backRightEncoderPositions.append(entry)
+
+                    out.write(entry + " ")
+                    i += 1
+                    if i == 6:
+                        i = 0
+                        out.write("\n")
+            else:
+                # print(line)
+                trash.write(line + " \n")
+    years = str(year)
+    frontLeftEncoderPositionArray = ",\n\t\t".join(frontLeftEncoderPositions)
+    frontRightEncoderPositionArray = ",\n\t\t".join(frontRightEncoderPositions)
+    backLeftEncoderPositionArray = ",\n\t\t".join(backLeftEncoderPositions)
+    backRightEncoderPositionArray = ",\n\t\t".join(backRightEncoderPositions)
+    if sadness < 1000:
```

Comment:
This would benefit from being split out into a separate function or method

Reply:
done